include/nbl/builtin/hlsl/subgroup/arithmetic_portability_impl.hlsl

```diff
@@ -0,0 +1,366 @@
+// Copyright (C) 2023 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_BUILTIN_HLSL_SUBGROUP_ARITHMETIC_PORTABILITY_IMPL_INCLUDED_
+#define _NBL_BUILTIN_HLSL_SUBGROUP_ARITHMETIC_PORTABILITY_IMPL_INCLUDED_
+
+#include "nbl/builtin/hlsl/glsl_compat/subgroup_arithmetic.hlsl"
+#include "nbl/builtin/hlsl/glsl_compat/subgroup_shuffle.hlsl"
+#include "nbl/builtin/hlsl/binops.hlsl"
+#include "nbl/builtin/hlsl/subgroup/ballot.hlsl"
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace subgroup
+{
+
+#ifdef NBL_GL_KHR_shader_subgroup_arithmetic
+namespace native
+{
+
+template<typename T, class Binop>
+struct reduction;
+template<typename T, class Binop>
+struct exclusive_scan;
+template<typename T, class Binop>
+struct inclusive_scan;
+
+// *** AND ***
+template<typename T>
+struct reduction<T, binops::bitwise_and<T> >
```

Comment:
I'm probably repeating myself, but does HLSL support partial specialization like this?

Reply:
oh yes it does!